config/common.go

```diff
@@ -141,3 +174,95 @@ func (cs *commonStore) validate(cfg *model.Config) error {
 
 	return nil
 }
+
+// removeEnvOverrides returns a new config without the current environment overrides.
+// If a config variable has an environment override, that variable is set to the value that was
+// read from the store.
+func (cs *commonStore) removeEnvOverrides(cfg *model.Config) *model.Config {
+	// When saving, iterate through the environmentOverrides map and check:
+	// foreach envOverrides, if config == loadedConfigNoEnv, then the environment override matched the original setting. No change
+	// foreach envOverrides, if config != loadedConfigNoEnv, then:
+	//    a) if config == loadedConfig, persist loadedConfigNoEnv (we don't want to persist the envSetSetting).
+	//    b) if config != loadedConfig, persist config (the user has subsequently changed the setting after the envOverride was applied)
+	//       NOTE: b cannot happen at the moment (from docs: "if a setting is configured through an environment variable,
+	//       modifying it in the System Console is disabled").
+	// Therefore, if config != loadedConfigNoEnv, always persist the loadedConfigNoEnv value
+
+	paths := getPaths(cs.environmentOverrides)
+	newCfg := cfg.Clone()
+	for _, path := range paths {
+		currentVal := getVal(newCfg, path)
+		loadedVal := getVal(cs.loadedConfigNoEnv, path)
+		if currentVal.Interface() != loadedVal.Interface() {
+			setVal(newCfg, path, loadedVal.Interface())
+		}
+	}
+	return newCfg
+}
+
+// getPaths is helper function for removeEnvOverrides
+func getPaths(m map[string]interface{}) [][]string {
+	return getPathsRec(m, nil, nil)
+}
+
+// getPathsRec is helper function for removeENvOverrides
+func getPathsRec(src interface{}, curPath []string, allPaths [][]string) [][]string {
+	if reflect.ValueOf(src).Kind() == reflect.Map {
+		for k, v := range src.(map[string]interface{}) {
+			allPaths = getPathsRec(v, append(curPath, k), allPaths)
+		}
+	} else {
+		allPaths = append(allPaths, curPath)
+	}
+	return allPaths
+}
+
+// getVal is helper function for removeEnvOverrides
+func getVal(src interface{}, path []string) reflect.Value {
+	var val reflect.Value
+	if reflect.ValueOf(src).Kind() == reflect.Ptr {
+		val = reflect.ValueOf(src).Elem().FieldByName(path[0])
+	} else {
+		val = reflect.ValueOf(src).FieldByName(path[0])
+	}
+	if val.Kind() == reflect.Ptr {
+		val = val.Elem()
+	}
+	if val.Kind() == reflect.Struct {
+		return getVal(val.Interface(), path[1:])
+	}
+	return val
+}
+
+// setVal is helper function for removeENvOverrides
+func setVal(tgt interface{}, path []string, newVal interface{}) {
```

Comment:
Could this method be replaced by a call to [Value.Set](https://golang.org/pkg/reflect/#Value.Set) instead?

Reply:
Yep. I could just eliminate that whole method because it was simplified so much. Thanks!